### PR TITLE
MatchError fix, value null is not a valid JSON (fixes #50)

### LINKED ISSUE #50: Empty YAML throws a MatchError
Parsing an empty YAML document causes the parser to exit with a `MatchError` exception pointing to `io.circe.yaml.parser.yamlToJson:97`.

This may be a bit tricky case, since empty file (in terms of fields - it may contain comments) is a valid YAML (with a single, `null` value), but not a valid JSON. Regardless, it seems reasonable to expect that a `ParsingFailure` would be returned in such case - instead, a `MatchError` is thrown.

Please let me know if you'd be willing to accept a pull request with fixes for that problem.

### PR DESCRIPTION
**Pull request by alisiikh**

### CODE REVIEW

**Comment by jeremyrsmith** [comment]
What does empty string currently evaluate to? AFAIK the spec says it should be valid YAML and is equivalent to "false".

**Comment by alisiikh** [comment]
Since circe-yaml first parses an empty string to JSON, the returned value is `null`, and during pattern matching there's no case for this which results in `scala.MatchError`.

**Comment by alisiikh** [comment]
I'm sorry, I confused the title and explanation a little bit. The snakeyaml parser returns `null`, and `null` is not a valid JSON but a valid YAML, so the code fails with `scala.MatchError`

**Comment by travisbrown** [comment]
@alisiikh I'm just now seeing this issue, but I think I'm with @jeremyrsmith—shouldn't this return `false` (as a `Json` value)?

**Comment by alisiikh** [comment]
@travisbrown yes, you're right it should be rather a false value than ParsingFailure.

File: src/main/scala/io/circe/yaml/parser/package.scala [hunk 1]
```diff
@@ -76,26 +76,29 @@ package object parser {
...
-        }.map(Json.fromJsonObject)
-      case sequence: SequenceNode =>
-        sequence.getValue.asScala.foldLeft(Either.right[ParsingFailure, List[Json]](List.empty[Json])) {
-          (arrEither, node) => for {
-            arr <- arrEither
-            value <- yamlToJson(node)
-          } yield value :: arr
-        }.map(arr => Json.fromValues(arr.reverse))
-      case scalar: ScalarNode => convertScalarNode(scalar)
+    if (node == null) {
```
Comment (travisbrown) [inline src/main/scala/io/circe/yaml/parser/package.scala]: Sorry to nitpick, but are there other cases where SnakeYAML will return `null` here?

Comment (alisiikh) [reply src/main/scala/io/circe/yaml/parser/package.scala]: As for my knowledge - for blank or empty strings, for `null` string it returns a `org.yaml.snakeyaml.nodes.ScalarNode`

The code below passes.

```scala
implicit def str2Reader(s: String): Reader = new StringReader(s)

val parser = new Yaml()
parser.compose("") shouldBe null
parser.compose("  ") shouldBe null
parser.compose("\n") shouldBe null
parser.compose("\r\n") shouldBe null
```

Passing null to the parser results in NullPointerException.

**Comment by travisbrown** [comment]
Okay, thanks, @alisiikh—I also just glanced at the docs and `null` seems to mean "no document" available, so this looks good to me.

**Pull request merged**